common/scala/src/main/scala/whisk/core/entity/WhiskAction.scala

```diff
@@ -238,7 +240,26 @@ object WhiskAction
                     Future.successful(action)
             }
         }
+    }
 
+    /**
+     *  utility function that given a fully qualified name for an action, resolve its possible package bindings and returns
+     *  the fully qualified name of the resolved action
+     */
+    def resolveAction(entityStore: EntityStore, fullyQualifiedName: FullyQualifiedEntityName)(
+        implicit ec: ExecutionContext, transid: TransactionId): Future[FullyQualifiedEntityName] = {
+        // first check that there is a package to be resolved
+        val entityPath = fullyQualifiedName.path
+        if (entityPath.defaultPackage) {
```

Comment:
The package will not be found and it will fail the future which is caught later on. There is no way around it, unless you see one. If we really want these things to typecheck we need to have Namespace, Package and EnitityName separately. And then an FQEN has all three. This requires an entire pass over the code which I'm not going to do now.